tests/experiment/test_run.py

```diff
@@ -595,25 +571,23 @@ def test_with_rare_categories():
         }
     }
 
-    for cat_cast in (True, False):
-        X_ = X.copy()
-        y_ = y.copy()
-        if cat_cast:
-            for c in X.columns:
-                X_[c] = X_[c].astype('category')
-            X_ = X_.iloc[:50, :]
-            y_ = y_.iloc[:50]
+    X_ = X.copy()
+    y_ = y.copy()
+    if cat_cast:
+        for c in X.columns:
+            X_[c] = X_[c].astype('category')
+        X_ = X_.iloc[:50, :]
+        y_ = y_.iloc[:50]
 
-        X_train, X_test, y_train, y_test = train_test_split(X_, y_, shuffle=False, test_size=0.5)
+    X_train, X_test, y_train, y_test = train_test_split(X_, y_, shuffle=False, test_size=0.5)
```

Comment:
`test_with_rare_categories` is flaky because `y_train` might end up having only one label and a model raises:

```
ValueError: Only one class present in y_true. ROC AUC score is not defined in that case.
```

https://github.com/nyanp/nyaggle/pull/53/checks?check_run_id=472872492#step:8:180

Reply:
Thanks, good catch 👍Adding `cv=StratifiedKFold(3, shuffle=True, random_state=0)` to `run_experiment` will solve the problem.

Reply:
BTW using the parameterized test is a great idea!

Reply:
Added `StratifiedKFold` and the tests passed!